FHEM/00_SIGNALduino.pm

```diff
@@ -3023,7 +3025,7 @@ SIGNALduino_Parse_MS($$$$%)
 	my $debug = AttrVal($iohash->{NAME},"debug",0);
 
 	
-	if (defined($clockidx) and defined($syncidx))
+	if (defined($clockidx) and defined($syncidx) and defined($hash->{msIdList}))
```

Comment:
@Ralf9 

Auf $hash->{msIdList} wird schon in SIGNALduino_Parse geprüft.
Das sieht jetzt aus, wie doppelt geprüft.

Reply:
Macht dies
if (@{$hash->{msIdList}}
dasselbe wie?
if (defined($hash->{msIdList})

Reply:
Mehr oder weniger,  @{$hash->{msIdList} kann 0 ergeben, dann ist die Bedingung nicht wahr und parse wird nicht aufgerufen

defined($hash->{msIdList} prüft dagegn nur ob msIdList definiert wurde.
Ich würde annehmen, habe es aber nicht geprüft, wenn @{$hash->{msIdList}  nicht in einen Fehler läuft, da $hash->{msIdList} ohnehin immer definiert ist, dann brauchen wir auf die Definition nicht zu prüfen, denn eine Liste mit 0 Einträgen ist auch definiert. Bei 0 Einträgen würden wir aber nie die define Prüfung aufrufen.